### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference [signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x627c26]

### ISSUE THREAD

**Issue by chrissound** [kubernetes/kubernetes#61712]
> /kind bug

```
kubectl delete -f deployment.resize.yaml                  
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x627c26]

goroutine 1 [running]:
panic(0x1bcc980, 0xc42000e100)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
k8s.io/kubernetes/pkg/kubectl.ReaperFor(0xc420277ec8, 0x4, 0xc420277f40, 0xa, 0x0, 0x0, 0x8, 0x1, 0x1, 0xc42084fc80)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/stop.go:95 +0xf26
k8s.io/kubernetes/pkg/kubectl/cmd/util.(*ring1Factory).Reaper(0xc4202def80, 0xc420b55500, 0xc420c49318, 0x2, 0x2, 0xc4203de120)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/util/factory_object_mapping.go:290 +0x151
k8s.io/kubernetes/pkg/kubectl/cmd/util.(*factory).Reaper(0xc4206e12c0, 0xc420b55500, 0x9, 0xc420c49378, 0x57e3db, 0xc42002efe8)
	<autogenerated>:92 +0x54
k8s.io/kubernetes/pkg/kubectl/cmd.ReapResult.func1(0xc420285300, 0x0, 0x0, 0x9, 0x650701)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go:252 +0xdb
k8s.io/kubernetes/pkg/kubectl/resource.ContinueOnErrorVisitor.Visit.func1(0xc420285300, 0x0, 0x0, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:348 +0x153
k8s.io/kubernetes/pkg/kubectl/resource.DecoratedVisitor.Visit.func1(0xc420285300, 0x0, 0x0, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:324 +0xe5
k8s.io/kubernetes/pkg/kubectl/resource.FlattenListVisitor.Visit.func1(0xc420285300, 0x0, 0x0, 0xc4209bf640, 0x442dd5)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:389 +0x57c
k8s.io/kubernetes/pkg/kubectl/resource.EagerVisitorList.Visit.func1(0xc420285300, 0x0, 0x0, 0x1c0, 0xc4204bff60)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:207 +0x153
k8s.io/kubernetes/pkg/kubectl/resource.(*StreamVisitor).Visit(0xc420a01080, 0xc4208b1b40, 0x2a21a00, 0xc4208b1bc0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:552 +0x308
k8s.io/kubernetes/pkg/kubectl/resource.(*FileVisitor).Visit(0xc4208b19e0, 0xc4208b1b40, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:501 +0x184
k8s.io/kubernetes/pkg/kubectl/resource.EagerVisitorList.Visit(0xc420277050, 0x1, 0x1, 0xc42084f2c0, 0x1, 0xc42084f2c0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:211 +0xf4
k8s.io/kubernetes/pkg/kubectl/resource.(*EagerVisitorList).Visit(0xc4208b1a80, 0xc42084f2c0, 0x0, 0x1edd562)
	<autogenerated>:122 +0x6e
k8s.io/kubernetes/pkg/kubectl/resource.FlattenListVisitor.Visit(0x2a1dac0, 0xc4208b1a80, 0xc420a01040, 0xc420a010c0, 0xc4208b1b01, 0xc420a010c0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:417 +0x94
k8s.io/kubernetes/pkg/kubectl/resource.(*FlattenListVisitor).Visit(0xc4208b1aa0, 0xc420a010c0, 0x18, 0x18)
	<autogenerated>:138 +0x6e
k8s.io/kubernetes/pkg/kubectl/resource.DecoratedVisitor.Visit(0x2a1db40, 0xc4208b1aa0, 0xc420277060, 0x2, 0x2, 0xc4208b1b00, 0xc420277001, 0xc4208b1b00)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:325 +0xd6
k8s.io/kubernetes/pkg/kubectl/resource.(*DecoratedVisitor).Visit(0xc42084f290, 0xc4208b1b00, 0x1c4d160, 0xc41fff693b)
	<autogenerated>:158 +0x78
k8s.io/kubernetes/pkg/kubectl/resource.ContinueOnErrorVisitor.Visit(0x2a1da40, 0xc42084f290, 0xc42012d880, 0x7f1908819000, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/visitor.go:352 +0xf1
k8s.io/kubernetes/pkg/kubectl/resource.(*ContinueOnErrorVisitor).Visit(0xc420277070, 0xc42012d880, 0x40ee98, 0x70)
	<autogenerated>:153 +0x65
k8s.io/kubernetes/pkg/kubectl/resource.(*Result).Visit(0xc42012d810, 0xc42012d880, 0x2ade0a8, 0x2a1da40)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/resource/result.go:96 +0x4e
k8s.io/kubernetes/pkg/kubectl/cmd.ReapResult(0xc42012d810, 0x2a3ff80, 0xc4206e12c0, 0x2a1cc40, 0xc42002e010, 0x2a20001, 0x0, 0xffffffffffffffff, 0x2a30000, 0x2a37cc0, ...)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go:277 +0x16f
k8s.io/kubernetes/pkg/kubectl/cmd.(*DeleteOptions).RunDelete(0xc4202c0160, 0x2a3ff80, 0xc4206e12c0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go:237 +0xc5
k8s.io/kubernetes/pkg/kubectl/cmd.NewCmdDelete.func1(0xc4202f9d40, 0xc4204bfea0, 0x0, 0x2)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/delete.go:143 +0x17c
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc4202f9d40, 0xc4204bef00, 0x2, 0x2, 0xc4202f9d40, 0xc4204bef00)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:603 +0x439
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc42022efc0, 0xc4204b5e90, 0x1, 0x1)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:689 +0x367
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(0xc42022efc0, 0xc4206e12c0, 0x2a1cc00)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:648 +0x2b
k8s.io/kubernetes/cmd/kubectl/app.Run(0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubectl/app/kubectl.go:39 +0xd5
main.main()
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubectl/kubectl.go:26 +0x22

```

**Environment**:
- Kubernetes version (use `kubectl version`):
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.4", GitCommit:"d6f433224538d4f9ca2f7ae19b252e6fcb66a3ae", GitTreeState:"clean", BuildDate:"2017-05-19T18:44:27Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"", Minor:"", GitVersion:"v1.9.0", GitCommit:"925c127ec6b946659ad0fd596fa959be43f0cc05", GitTreeState:"clean", BuildDate:"2018-01-26T19:04:38Z", GoVersion:"go1.9.1", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: Minikube 
- OS (e.g. from /etc/os-release):
```
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
ID_LIKE=archlinux
ANSI_COLOR="0;36"
HOME_URL="https://www.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://bugs.archlinux.org/"
```
- Kernel (e.g. `uname -a`):
`Linux archamd 4.14.29-1-lts #1 SMP Wed Mar 21 17:41:42 CET 2018 x86_64 GNU/Linux`

**deployment.resize.yaml**

    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: "resize-deployment"
      labels:
        appId: "resize-deployment"
    spec:
      replicas: 1
      selector:
        matchLabels:
          appId: "resize-deployment"
      template:
        metadata:
          labels:
            appId: "resize-deployment"
        spec:
          containers:
          - name: debian
            image: debian:stretch-slim
          volumes:
          - name: mysql-persistent-storage
            persistentVolumeClaim:
              claimName: testingchris3

**Comment by mountkin** [comment]
Duplicate of #61320 
Upgrade kubectl should fix the problem

**Comment by liggitt** [comment]
this is fixed in 1.8+

/close
